--- sendmail-8.14.4/sendmail/main.c.orig	2014-06-06 21:47:39.491335044 -0700
+++ sendmail-8.14.4/sendmail/main.c	2014-06-06 22:00:29.902135894 -0700
@@ -2607,13 +2607,13 @@
 #if NETINET
 		  case AF_INET:
 			(void) sm_snprintf(pbuf, sizeof(pbuf), "%d",
-					   RealHostAddr.sin.sin_port);
+					   ntohs(RealHostAddr.sin.sin_port));
 			break;
 #endif /* NETINET */
 #if NETINET6
 		  case AF_INET6:
 			(void) sm_snprintf(pbuf, sizeof(pbuf), "%d",
-					   RealHostAddr.sin6.sin6_port);
+					   ntohs(RealHostAddr.sin6.sin6_port));
 			break;
 #endif /* NETINET6 */
 		  default:
